### ISSUE TITLE
Feature suggestion: two rand seeds for weights initialization and training data position

### ISSUE THREAD

**Issue by martinpopel** [tensorflow/tensor2tensor#629]
It could be useful to be able to set separately the random seed for weights initialization and the random seed for initial training data position. Both should accept 0 meaning None, i.e. initialize the seed from system time. I would vote for keeping the default rand seed fixed (non-zero) in both cases.

**Comment by rsepassi** [comment]
I'd be concerned about knowing that you have to set both to get deterministic behavior. Also, not immediately clear to me how to go about doing this.

**Comment by martinpopel** [comment]
> I'd be concerned about knowing that you have to set both to get deterministic behavior.

Both would be set to a non-zero value by default, which means both would be deterministic.

> Also, not immediately clear to me how to go about doing this.

I admit I cannot find an easy way. Maybe making a wrapper object, which keeps the random state and sets it with `random.setstate()` before any call to `random.randint()` or `random.random()` etc. and updates the stored state afterwards with `random.getstate()`, but this seems too hacky and error prone in case of concurrency.

As I think about it, it would be much better to just provide an option which turns off the [skipping of random portion of training data](https://github.com/tensorflow/tensor2tensor/blob/40758df26f92cdaa20869ef9d470da997ad89557/tensor2tensor/data_generators/problem.py#L764).
I would even suggest to use this option by default unless the `schedule=continuous_train_and_eval`.
My final goal would be to allow users to turn off training data shuffling and skipping completely, but this is a bit more difficult, see #741.
(I am too busy to implement it now and send a PR. I can live with the workaround with one datashard. So feel free to close both the issues if there is no one interested in implementing it.)
